src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        if (this.factory == null) {
+            throw new DataProcessingException("Factory not initialized");
```

Comment:
Checking if the factory is initialized inside the `add` method is unnecessary because the factory should be initialized during the construction of the DAO instance. If the factory was not properly initialized, the DAO should not exist in the first place.